### DISCUSSION TITLE
Support Svelte 4

### DISCUSSION THREAD

**Discussion by vekunz** [TanStack/virtual#601] [General]
Svelte 4 has been out for a while now. Any dependencies we have in our projects support Svelte 4, but only TanStack Virtual has no Svelte 4 support. Since `svelte-virtual` has a peerDependency on Svelte 3, we cannot upgrade to Svelte 4. It would be great if `svelte-virtual` supports Svelte 4 also.

**Comment by piecyk** [comment]
@vekunz can you help with that, update peerDependency and check if everything works?

**Reply by vekunz** [reply]
I could check if our use case works. But I don't have the time to check, if "everything" works.

**Reply by piecyk** [reply]
That would be awesome, i think if your use case works then we are good to go!

**Reply by vekunz** [reply]
Sorry for the delay, it has been busy lately. Our (very basic) use case works great with Svelte 4.

We have list items with a static height, dynamic height of the list itself, and very dynamic content in the list.

**Comment by MarcusVirg** [comment]
I can probably help out here if need be. I have also tested my use-case using Svelte 4 and it all seems to work. @piecyk Would you be opposed if I made a PR updating the peerDependency to support Svelte 3 & 4? I can also include Svelte examples in this PR since they seem to be non-existent.

**Reply by MarcusVirg** [reply]
Okay a PR has been created with most of the examples for Svelte:
https://github.com/TanStack/virtual/pull/624
